exporter/stats/stackdriver/stackdriver_test.go

```diff
@@ -34,6 +34,42 @@ import (
 	"google.golang.org/grpc/status"
 )
 
+func TestRejectBlankProjectID(t *testing.T) {
+	ids := []string{"", "     ", " "}
+	for _, projectID := range ids {
+		opts := Options{ProjectID: projectID}
+		exp, err := NewExporter(opts)
+		if err == nil || exp != nil {
+			t.Errorf("%q ProjectID must be rejected: NewExporter() = %v err = %q",
+				projectID, exp, err)
+		}
+	}
+}
+
+// Ensure only one exporter per projectID per process, any
+// subsequent invocations of NewExporter should fail.
+func TestNewExporterSingletonPerProcess(t *testing.T) {
+	ids := []string{"open-census.io", "x", "fakeProjectID"}
+	for _, projectID := range ids {
+		opts := Options{ProjectID: projectID}
+		exp, err := NewExporter(opts)
+		if err != nil {
+			t.Errorf("NewExporter() projectID = %q err = %q", projectID, err)
+			continue
+		}
+		if exp == nil {
+			t.Errorf("NewExporter returned a nil Exporter")
+			continue
+		}
+		for i := 0; i < 10; i++ {
```

Comment:
You don't need to iterate for 10 times.

Just try to call NewExporter once more.